source/topic3.rst

```diff
@@ -208,16 +208,232 @@ Getters and Setters
+    }
+
+* Notice how we actually use the string's `equals` too
+* In the above example, using ``this`` is not necessary, but may be helpful in eliminating ambiguity
+
+.. warning::
+
+    The example ``equals()`` included in the ``Friend.java`` file is a little more complex,
+    but is a more complete ``equals``
+
+
+Creating an Instance of a Friend
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: java
+    :linenos:
+
+    public class SomeClass {
+        public static void main(String[] args) {
+
+            // Declare a Friend variable
+            // Create an instance of a Friend
+            // Assign the variable to reference the newly created Friend
+            Friend aFriend = new Friend("Bob", "Smith", "bsmith@gmail.com");
+
+        }
+    }
+
+* Here's an example of us actually creating an instance of the ``Friend`` class
+* This is a really really really simple example where we create it and do nothing with it
+
+* There is a bit going on here:
+    * Declare a ``Friend`` variable
+        * ``Friend aFriend``
+    * Create an instance of a ``Friend``
+        * ``new Friend("Bob", "Smith", "bsmith@gmail.com")``
+    * Assign the variable to reference the newly created object
+        * The equals bit
+        * We have to be careful about what is actually stored in the ``aFriend``
+
+        .. image:: ../img/reference_aFriend.png
+           :width: 250 px
+           :align: center
+
+
+
+* If I were to write another line of code like this
+    * ``Friend bFriend = new Friend("Jane", "Doe", "jdoe@gmail.com");``
+* ``aFriend`` would have a ``firstName`` of Bob
+* ``bFriend`` has a ``firstName`` of Jane
+* They both have the ``firstName`` field, but the actual value associated with it differs
+
+    .. image:: ../img/reference_aFriend_bFriend.png
+       :width: 500 px
+       :align: center
+
+
+* Below is an example of two``Friend`` objects being created and being used
+    * We get ``aFriend``'s first name
+    * Use the ``toString`` method
+    * Use the ``equals`` method
+
+.. code-block:: java
+    :linenos:
+
+    Friend aFriend = new Friend("Bob", "Smith", "bsmith@gmail.com");
+    Friend bFriend = new Friend("Jane", "Doe", "jdoe@gmail.com");
+
+    System.out.println(aFriend.getFirstName());
+    System.out.println(aFriend);
+    System.out.println(bFriend);
+    System.out.println(aFriend.equals(bFriend));
+
+
+* What do you think the output of this would be?
+
+References
+^^^^^^^^^^
+
+* We have to be careful about what is actually stored in these variables
+* We're not storing the object in that variable --- we store a reference to where the object is in memory
+
+.. code-block:: java
+    :linenos:
+    :emphasize-lines: 4
+
+    Friend aFriend = new Friend("Bob", "Smith", "bsmith@gmail.com");
+    Friend bFriend = new Friend("Jane", "Doe", "jdoe@gmail.com");
+
+    bFriend = aFriend;
+
+
+* In the above example, ``bFriend = aFriend`` copies the contents of ``aFriend`` and puts the copy in ``bFriend``
```

Comment:
_contents_ is potentially ambiguous here. Maybe editing the images to include **addr of Bob** in the boxes would make it clearer that the _contents_ is the virtual memory address, and not the class instance itself.

Reply:
<3